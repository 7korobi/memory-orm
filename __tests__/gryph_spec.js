require('./models/index')
const Mem = require('../lib/index')

describe('gryph', function () {
  test('points snapshot', function () {
    expect(Mem.Query.points.reduce).toMatchSnapshot()
  })

  test('point structure snapshot', function () {
    expect(Mem.Query.points.pluck('id', 'x', 'y')).toMatchSnapshot()
  })

  test('lines snapshot', function () {
    expect(Mem.Query.lines.reduce).toMatchSnapshot()
  })

  test('deltas snapshot', function () {
    expect(Mem.Query.deltas.reduce).toMatchSnapshot()
  })
})
