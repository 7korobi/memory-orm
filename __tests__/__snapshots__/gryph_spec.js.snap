// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gryph deltas snapshot 1`] = `
Object {
  "hash": Object {},
  "list": list [],
  "set": Array [],
}
`;

exports[`gryph lines snapshot 1`] = `
Object {
  "hash": Object {
    "1,-1*-1,1": Line [
      Array [
        1,
        -1,
      ],
      Array [
        -1,
        1,
      ],
    ],
    "1,1*-1,-1": Line [
      Array [
        1,
        1,
      ],
      Array [
        -1,
        -1,
      ],
    ],
    "1,1*1,-1*-1,1*-1,-1": Line [
      Array [
        1,
        1,
      ],
      Array [
        1,
        -1,
      ],
      Array [
        -1,
        1,
      ],
      Array [
        -1,
        -1,
      ],
    ],
  },
  "list": list [
    Line [
      Array [
        1,
        1,
      ],
      Array [
        -1,
        -1,
      ],
    ],
    Line [
      Array [
        1,
        -1,
      ],
      Array [
        -1,
        1,
      ],
    ],
    Line [
      Array [
        1,
        1,
      ],
      Array [
        1,
        -1,
      ],
      Array [
        -1,
        1,
      ],
      Array [
        -1,
        -1,
      ],
    ],
  ],
  "set": Array [
    "1,1*-1,-1",
    "1,-1*-1,1",
    "1,1*1,-1*-1,1*-1,-1",
  ],
}
`;

exports[`gryph point structure snapshot 1`] = `
list [
  Array [
    "1*1",
    1,
    1,
  ],
  Array [
    "1*-1",
    1,
    -1,
  ],
  Array [
    "-1*1",
    -1,
    1,
  ],
  Array [
    "-1*-1",
    -1,
    -1,
  ],
]
`;

exports[`gryph points snapshot 1`] = `
Object {
  "hash": Object {
    "-1*-1": struct [
      -1,
      -1,
    ],
    "-1*1": struct [
      -1,
      1,
    ],
    "1*-1": struct [
      1,
      -1,
    ],
    "1*1": struct [
      1,
      1,
    ],
  },
  "list": list [
    struct [
      1,
      1,
    ],
    struct [
      1,
      -1,
    ],
    struct [
      -1,
      1,
    ],
    struct [
      -1,
      -1,
    ],
  ],
  "set": Array [
    "1*1",
    "1*-1",
    "-1*1",
    "-1*-1",
  ],
}
`;
